test/robot/job/worker/action.go

```diff
@@ -95,12 +95,13 @@ func (a *Actions) apply(ctx context.Context, item Action) error {
 
 func (a *Actions) do(ctx context.Context, w *Worker, input Input) (string, error) {
 	a.mu.Lock()
-	defer a.mu.Unlock()
 	action := proto.Clone(a.nullAction).(Action)
 	action.Init(id.Unique().String(), input, w.Info)
 	if err := a.ledger.Add(ctx, action); err != nil {
 		return "", err
 	}
+	// unlock here so other do calls can be invoked after adding to ledger
```

Comment:
ok the latest version has a local function to handle defer correctly. If you prefer the separate private function let me know.